src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/ComplexTypeModelBinder.cs

```
@@ -135,7 +148,13 @@ protected virtual bool CanBindProperty(ModelBindingContext bindingContext, Model
         /// </returns>
         protected virtual Task BindProperty(ModelBindingContext bindingContext)
         {
-            return bindingContext.OperationBindingContext.ModelBinder.BindModelAsync(bindingContext);
+            IModelBinder binder;
+            if (_propertyBinders.TryGetValue(bindingContext.ModelMetadata, out binder))
```

yeah, this should fail hard.